packages/app/src/systems/Pool/hooks/usePreviewAddLiquidity.ts

```diff
@@ -18,11 +18,15 @@ export interface UsePreviewAddLiquidityProps {
 
 export function usePreviewAddLiquidity({ fromInput, poolInfo }: UsePreviewAddLiquidityProps) {
   const { totalLiquidity, poolTokensNum } = useUserPositions();
+
   const liquidityFactor = (fromInput.amount || ZERO) * toBigInt(totalLiquidity);
   const previewTokens = divideFnValidOnly(liquidityFactor, totalLiquidity);
   const nextTotalTokenSupply = previewTokens + toNumber(poolInfo?.lp_token_supply || BigInt(0));
-  const nextCurrentPoolShare =
-    divideFnValidOnly(BigInt(previewTokens) + poolTokensNum, BigInt(nextTotalTokenSupply)) || 1;
```

Comment:
The `|| 1` here is used on liquidity creation to show 100% as the user's current position. Did you test it in this state?

Reply:
Yes, is returning 100% when the user has 100% of the pool and 0 when he has nothing.